sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-file-handler.ts

```diff
@@ -0,0 +1,224 @@
+      (disbursement) => disbursement.id,
+    );
+
+    //Create records and create the unique file sequence number.
+    let uploadResult: ECertUploadResult;
+    const now = new Date();
+    const dayOfTheYear = getDayOfTheYear(now);
+    await this.sequenceService.consumeNextSequence(
+      sequenceGroup,
+      async (nextSequenceNumber: number, entityManager: EntityManager) => {
+        try {
+          this.logger.log(
+            `Creating  ${offeringIntensity} e-Cert file content...`,
+          );
+          const fileContent = eCertIntegrationService.createRequestContent(
+            disbursementRecords,
+            nextSequenceNumber,
+          );
+
+          // Create the request filename with the file path for the e-Cert File.
+          const fileInfo = await this.createRequestFileName(
+            `${fileCode}${now.getFullYear()}${dayOfTheYear}`,
+            nextSequenceNumber,
+          );
+
+          // Creates the repository based on the entity manager that
+          // holds the transaction already created to manage the
+          // sequence number.
+          const disbursementScheduleRepo =
+            entityManager.getRepository(DisbursementSchedule);
+          await this.disbursementScheduleService.updateRecordsInSentFile(
+            disbursementIds,
+            now,
+            disbursementScheduleRepo,
+          );
+
+          this.logger.log(`Uploading ${offeringIntensity} content...`);
+          await eCertIntegrationService.uploadContent(
+            fileContent,
+            fileInfo.filePath,
+          );
+
+          uploadResult = {
+            generatedFile: fileInfo.filePath,
+            uploadedRecords: disbursementRecords.length,
+          };
+        } catch (error) {
+          this.logger.error(
+            `Error while uploading content for ${offeringIntensity} e-Cert file: ${error}`,
+          );
+          throw error;
+        }
+      },
+    );
+    return uploadResult;
+  }
+
+  /**
+   * Create the e-Cert record with the information needed to generate the
+   * entire record to be sent to ESDC.
+   * @param disbursement disbursement that contains all information to
+   * generate the record.
+   * @returns e-Cert record.
+   */
+  private createECertRecord(disbursement: DisbursementSchedule): ECertRecord {
+    const now = new Date();
+    const application = disbursement.studentAssessment.application;
+    const [addressInfo] = application.student.contactInfo.addresses;
+    const fieldOfStudy = getFieldOfStudyFromCIPCode(
+      application.currentAssessment.offering.educationProgram.cipCode,
+    );
+    const awards = disbursement.disbursementValues.map(
+      (disbursementValue) =>
+        ({
+          valueType: disbursementValue.valueType,
+          valueCode: disbursementValue.valueCode,
+          valueAmount: disbursementValue.valueAmount,
+        } as Award),
+    );
+
+    return {
+      sin: application.student.sin,
+      courseLoad: 50,
```

Comment:
will it always be `50`

Reply:
No, this is hardcoaded as we do not have this value in our database, this will be changed. Added a TODO comment for better understanding.

Reply:
It is temporary. Please add a TODO @guru-aot.